prusti-viper/src/encoder/mir/procedures/encoder/mod.rs

```diff
@@ -424,116 +427,277 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         Ok(())
     }
 
-    fn encode_lft(
+    fn encode_lft_for_statement(
         &mut self,
         block_builder: &mut BasicBlockBuilder,
         location: mir::Location,
         original_lifetimes: &mut BTreeSet<String>,
         derived_lifetimes: &mut BTreeMap<String, BTreeSet<String>>,
     ) -> SpannedEncodingResult<()> {
-        let (new_lifetimes, ended_lifetimes, introduced_derived_lifetimes) =
-            self.update_lifetimes(original_lifetimes, derived_lifetimes, location);
-        self.encode_end_lft(block_builder, location, ended_lifetimes)?;
-        self.encode_new_lft(block_builder, location, new_lifetimes)?;
-        self.encode_lft_assignments(block_builder, location, introduced_derived_lifetimes)?;
+        let new_derived_lifetimes = self.lifetimes.get_origin_contains_loan_at_mid(location);
+        block_builder.add_comment(format!("Prepare lifetimes for statement {:?}", location));
+        self.encode_lft(
+            block_builder,
+            location,
+            original_lifetimes,
+            derived_lifetimes,
+            &new_derived_lifetimes,
+        )?;
         Ok(())
     }
 
-    fn encode_end_lft(
+    /// Adds all statements needed for the encoding of the location.
+    fn encode_lft(
```

Comment:
This file is becoming huge. Could you move all lifetime related methods to a separate file `prusti-viper/src/encoder/mir/procedures/encoder/lifetimes.rs`?

Reply:
yes, I added a `LifetimesEncoder` trait in a seperate file.